airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/objects/ConnectorSpecification.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol.objects;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.airbyte.commons.protocol.objects.serde.ConnectionSpecificationDeserializer;
+import io.airbyte.commons.protocol.objects.serde.ConnectionSpecificationSerializer;
+import io.airbyte.commons.protocol.objects.serde.JsonSerializable;
+import io.airbyte.protocol.models.AdvancedAuth;
+import io.airbyte.protocol.models.AuthSpecification;
+import java.net.URI;
+import java.util.List;
+
+@JsonDeserialize(using = ConnectionSpecificationDeserializer.class)
```

Comment:
should these be declared on the adapters? E.g. if the platform is on protocol v42, but an older connector still uses protocol v41, and it has a call to `Jsons.serialize(new ConnectorSpecificationAdapterV41(...))`